### PR TITLE
Timer interrupt ISR 

### LINKED ISSUE #37: Create timer interrupt
We may need to create an ISR or two. In principle it is good practice.

### PR DESCRIPTION
**Pull request by jamie-54**
## Main Changes

implementation of a timer ISR to guarantee that the main loop on the ESP32 runs at a fixed rate.

## Associated Issues

- Closes #37

## Tests

This has been tested on the quadcopter

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -80,111 +118,110 @@ void setGainsFromRcTrimmers()
...
+  timerState = !timerState;
+  taskEXIT_CRITICAL_ISR(&timerMux);
+
   float quaternionImuData[4];
   float measuredAngularVelocity[3];
   float angularVelocityCorrected[3];
 
-  // if raspberryEsp32Interface data received update the last data read time.
-  if (raspberryEsp32Interface.readPiData())
+  if (timerState==true)
```
Comment (alphaville) [inline src/main.cpp:129]: To avoid this lengthy if-block you can write 

```c++
if (!timerState) return;
```

**Review by alphaville — APPROVED** [review]

File: src/main.cpp (hunk 1 above)
Comment (jamie-54) [reply src/main.cpp:129]: Sounds good, does it matter if its at start or end?

Comment (alphaville) [reply src/main.cpp:129]: It should be at the start

**Comment by jamie-54** [comment]
Can this be merged now?

**Pull request merged**